packages/react-components/src/components/MentionPopover.tsx

```diff
@@ -0,0 +1,322 @@
+  /* @conditional-compile-remove(mention) */
+  const ids = useIdentifiers();
+  const localeStrings = useLocale().strings;
+  const popoverRef = useRef() as React.MutableRefObject<HTMLDivElement>;
+
+  const [position, setPosition] = useState<Position>({ left: 0 });
+  const [hoveredSuggestion, setHoveredSuggestion] = useState<Mention | undefined>(undefined);
+  const [changedSelection, setChangedSelection] = useState<boolean | undefined>(undefined); // Selection UI as per teams
+
+  const dismissPopoverWhenClickingOutside = useCallback(
+    (e: MouseEvent) => {
+      const target = e.target as HTMLElement;
+      if (popoverRef.current && !popoverRef.current.contains(target)) {
+        onDismiss && onDismiss();
+      }
+    },
+    [onDismiss]
+  );
+
+  useEffect(() => {
+    if (changedSelection === undefined) {
+      setChangedSelection(false);
+    } else if (changedSelection === false) {
+      setChangedSelection(true);
+    }
+  }, [activeSuggestionIndex, changedSelection]);
+
+  useEffect(() => {
+    window && window.addEventListener('click', dismissPopoverWhenClickingOutside);
+    return () => {
+      window && window.removeEventListener('click', dismissPopoverWhenClickingOutside);
+    };
+  }, [dismissPopoverWhenClickingOutside]);
+
+  // Determine popover position
+  useEffect(() => {
+    const rect = target?.current?.getBoundingClientRect();
+    const maxWidth = 200;
+    const finalPosition: Position = { maxWidth };
+
+    // Figure out whether it will fit horizontally
+    const leftOffset = targetPositionOffset?.left ?? 0;
+    if (leftOffset + maxWidth > (rect?.width ?? 0)) {
+      finalPosition.right = (rect?.width ?? 0) - leftOffset;
+    } else {
+      finalPosition.left = leftOffset;
+    }
+
+    if (location === 'below') {
+      finalPosition.top = (rect?.height ?? 0) + (targetPositionOffset?.top ?? 0);
+    } else {
+      // (location === 'above')
+      finalPosition.bottom = (rect?.height ?? 0) - (targetPositionOffset?.top ?? 0);
+    }
+    setPosition(finalPosition);
+  }, [location, target, targetPositionOffset]);
+
+  const handleOnKeyDown = useCallback(
+    (e: React.KeyboardEvent<HTMLDivElement>) => {
+      switch (e.key) {
+        case 'Escape':
+          onDismiss && onDismiss();
+          break;
+        default:
+          break;
+      }
+    },
+    [onDismiss]
+  );
+
+  const personaRenderer = (displayName?: string): JSX.Element => {
+    const displayNamePlaceholder = localeStrings.participantItem.displayNamePlaceholder;
+    const avatarOptions = {
+      text: displayName?.trim() || displayNamePlaceholder,
+      size: PersonaSize.size24,
+      initialsColor: theme.palette.neutralLight,
+      initialsTextColor: theme.palette.neutralSecondary,
+      showOverflowTooltip: false,
+      showUnknownPersonaCoin: !displayName?.trim() || displayName === displayNamePlaceholder
+    };
+
+    return <Persona {...avatarOptions} />;
+  };
+
+  const defaultOnRenderSuggestionItem = (
```

Comment:
This could be a function living outside of the react function, so it wouldn't be created everytime